File: src/main/java/com/mastery/java/task/model/Employee.java
package com.mastery.java.task.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import javax.persistence.*;
import javax.validation.constraints.*;
import java.math.BigDecimal;
import java.time.LocalDate;


@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "employees")
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long employeeId;

    @NotNull(message = "First name cannot be null")
    private String firstName;

    @NotNull(message = "Last name cannot be null")
    private String lastName;

    @Min(1)
    @NotNull(message = "Department cannot be null")
    private Integer departmentId;

    @NotNull(message = "Job title cannot be null")
    @Size(min = 2, max = 50, message = "Job title size must be between 2 and 20 symbols")
    private String jobTitle;

    @NotNull
    @Enumerated(EnumType.STRING)
    private Gender gender;

    @NotNull
    @JsonDeserialize(using = LocalDateDeserializer.class)
    @JsonSerialize(using = LocalDateSerializer.class)
    private LocalDate dateOfBirth;


// TODO: It using for schema-migration with help liquibase
//
    @NotNull(message = "Salary cannot be less than 400$")
    @Min(400)
    private BigDecimal salary;

}

File: frontend/src/app/model/Employee.ts
import {Gender} from './Gender';

export class Employee {

    employeeId: number;
    firstName: string;
    lastName: string;
    departmentId?: number;
    jobTitle: string;
    gender: Gender;
    dateOfBirth?: string;
    salary?: number;


    constructor(firstName: string,
                lastName: string,
                jobTitle: string,
                gender: Gender,
                dateOfBirth?: string,
                departmentId?: number,
                salary?: number)
    {
        this.firstName = firstName;
        this.lastName = lastName;
        this.departmentId = departmentId;
        this.jobTitle = jobTitle;
        this.gender = gender;
        this.dateOfBirth = dateOfBirth;
        this.salary = salary;
    }
}
